leader-election-impl/src/main/java/com/palantir/leader/proxy/AwaitingLeadershipProxy.java

```diff
@@ -167,15 +161,16 @@ private RuntimeException handleDelegateThrewException(
         Throwable cause = exception.getCause();
         if (cause instanceof SuspectedNotCurrentLeaderException) {
             handleSuspectedNotCurrentLeader(leadershipToken, (SuspectedNotCurrentLeaderException) cause);
-        }
-        if (cause instanceof ServiceNotAvailableException || cause instanceof NotCurrentLeaderException) {
-            leadershipStateManager.invalidateStateOnLostLeadership(leadershipToken, cause);
-        }
-        // Prevent blocked lock requests from receiving a non-retryable 500 on interrupts
-        // in case of a leader election.
-        if (cause instanceof InterruptedException && !leadershipCoordinator.isStillCurrentToken(leadershipToken)) {
-            throw leadershipCoordinator.notCurrentLeaderException(
-                    "received an interrupt due to leader election.", cause);
+        } else if (cause instanceof ServiceNotAvailableException) {
```

Comment:
Thanks! Would you mind leaving a comment + adding a test that covers this code path when thrown a NotCurrentLeaderException? Just want to make sure we don't regress if we change the parent type of the exception - and I couldn't see one where we had a test covering that...